x-pack/qa/rolling-upgrade/build.gradle

```diff
@@ -142,6 +142,11 @@ for (Version bwcVersion : bwcVersions.wireCompatible) {
     nonInputProperties.systemProperty('tests.rest.cluster', "${-> testClusters."${baseName}".allHttpSocketURI.join(",")}")
     nonInputProperties.systemProperty('tests.clustername', "${-> testClusters."${baseName}".getName()}")
     systemProperty 'tests.rest.suite', 'upgraded_cluster'
+    if (bwcVersion.onOrAfter("8.0.0")) {
+      systemProperty 'tests.rest.blacklist', [
+        'upgraded_cluster/100_geo_shape_doc_values/Test upgraded 8.x cluster with 8.x and 7.x geo_shape fields'
```

Comment:
Hey @rjernst. I do not like what I'm doing here, but I did it to express my intention to 
help reach a better solution. I was a bit surprised that the rolling-upgrade tests were run on 8.0 -> 8.0 
and I do not want my tests to be run in this scenario. To achieve this I used `skip:` and this here.

Do you think that my tests do not belong in `qa/rolling-upgrade` or do you think we need a clearer way to have tests that only run on true upgrade scenarios?

Reply:
The purpose of 8.0 -> 8.0 is to test rolling restart. While I agree this should be split out into it's own dedicated test (and so does core/infra in general; we've discussed it a few times), can you further explain the reason this can't work? The test is modeled this way currently not only because it was convenient, but also because rolling restart and upgrade are in most cases exactly the same, just differing in which old version they are talking to, but still expecting consistent behavior during the mixed/upgraded state from the features being tested.

Reply:
The idea is to verify the behavior of users indexing into an existing index that does not have geo-shape fields with doc-value support.

I can achieve this the same in 8.0 by explicitly setting `doc_values: false`, but this is not something I can do in 7.x at the moment. Testing a mix of old indices and new indices is not possible in an 8.0 -> 8.0 setup. So, if that is the case, does that mean this is not something we should be explicitly testing? and just assume a known behavior?

Reply:
Can this not be done with 7.x "at the moment" because this change needs to be backported? If so, can you build the test here, and test against a local 7.x with this backported? Since this will be backported and there is backcompat logic, wouldn't the bwc tests need to be disabled anyways in this PR?

Reply:
that is true, but then the same will be the case with 7.8 -> 7.8.

This test will likely be backported and removed from master, but kept in 7.x

Reply:
In either case, couldn't you use the version skipping feature already present in yaml tests?

Reply:
no, because the `upgraded_cluster` test will always be run on the latest version, no way knowing whether it came from the same version or a previous version. I used skips for the old_cluster and the mixed_cluster tests.

Reply:
I've changed the upgraded_cluster tests to fit into the framework of the tests. thanks Ryan